src/update/UpdateManager.cpp

```diff
@@ -156,16 +161,41 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
                                     nullptr,    // parent
                                     wxPD_APP_MODAL | wxPD_ELAPSED_TIME | wxPD_SMOOTH));
 
+                                const wxPlatformInfo& info = wxPlatformInfo::Get();
+                                if (info.GetOperatingSystemId() & wxOS_WINDOWS)
+                                    audacityPatchFilename += ".exe";
+                                else if(info.GetOperatingSystemId() & wxOS_MAC)
+                                    audacityPatchFilename += ".dmg";
+
                                 auto audacityInstallerPath = wxFileName(
                                     wxStandardPaths::Get().GetUserDir(wxStandardPaths::Dir_Downloads)
                                     + FileNames::GetPathSeparator()
                                     + audacityPatchFilename);
+
+                                mAudacityInstaller.open(audacityInstallerPath.GetFullPath().ToStdString(), std::ios::binary);
                                 
                                 });
 
                             mProgressDialog->Pulse();
                         }
                     );
+
+                    downloadResponse->setOnDataReceivedCallback([downloadResponse, this](audacity::network_manager::IResponse*) {
+                        if (downloadResponse->getError() == audacity::network_manager::NetworkError::NoError)
+                        {
+                            std::vector<char> buffer(downloadResponse->getBytesAvailable());
+
+                            size_t bytes = downloadResponse->readData(buffer.data(), buffer.size());
+                            //NOTE need add error checkpoints: if (bytes != downloadResponse->getBytesAvailable())
```

Comment:
Why not do this error checking?

Reply:
Because it can successfully read in next callback cycle, may be need additional check all readed bytes in `FinishCallback` also.